packages/router-plugin/src/router.state.ts

```diff
@@ -107,12 +113,8 @@ export class RouterState {
     });
   }
 
```

Comment:
```ts
this._router.events.subscribe(e => {
  if (RouterState.checkRouterEvent(e, [RoutesRecognized])) {
    ..
  } else if (RouterState.checkRouterEvent(e, [GuardsCheckEnd, ResolveEnd])) {
  ..
  } else if (RouterState.checkRouterEvent(e, [NavigationEnd]) {
  ..
  }
})

private static checkRouterEvent(e: RouterEvent, eventTypes: RouterEvent[]) {
  return events.some(event => e incenceof event);
}
```

Reply:
what do you mean?

Reply:
@splincode I think that your suggested code makes things more complicated for not much gain.

Reply:
@splincode The change you are suggesting makes things more complicated for very little benefit. It doesn't reduce repetition either. The existing code is easy to understand and can be optimised by the JS engine too.